tick/eval/evaluation_funcs.go

```diff
@@ -0,0 +1,788 @@
+			return boolFalseResultContainer, err
+		}
+
+		return resultContainer{BoolValue: left <= right, IsBoolValue: true}, nil
+
+	},
+
+	operationKey{operator: tick.TokenEqual, leftType: TString, rightType: TString}: func(scope *tick.Scope, executionAux ExecutionAux, leftNode, rightNode NodeValueAccessor) (resultContainer, error) {
+		var left string
+		var right string
+		var err error
+
+		if left, err = leftNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		if right, err = rightNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		return resultContainer{BoolValue: left == right, IsBoolValue: true}, nil
+
+	},
+
+	operationKey{operator: tick.TokenNotEqual, leftType: TString, rightType: TString}: func(scope *tick.Scope, executionAux ExecutionAux, leftNode, rightNode NodeValueAccessor) (resultContainer, error) {
+		var left string
+		var right string
+		var err error
+
+		if left, err = leftNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		if right, err = rightNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		return resultContainer{BoolValue: left != right, IsBoolValue: true}, nil
+
+	},
+
+	operationKey{operator: tick.TokenRegexNotEqual, leftType: TString, rightType: TRegex}: func(scope *tick.Scope, executionAux ExecutionAux, leftNode, rightNode NodeValueAccessor) (resultContainer, error) {
+		var left string
+		var right *regexp.Regexp
+		var err error
+
+		if left, err = leftNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		if right, err = rightNode.RegexValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		return resultContainer{BoolValue: !right.MatchString(left), IsBoolValue: true}, nil
+
+	},
+
+	operationKey{operator: tick.TokenRegexEqual, leftType: TString, rightType: TRegex}: func(scope *tick.Scope, executionAux ExecutionAux, leftNode, rightNode NodeValueAccessor) (resultContainer, error) {
+		var left string
+		var right *regexp.Regexp
+		var err error
+
+		if left, err = leftNode.StringValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		if right, err = rightNode.RegexValue(scope, executionAux); err != nil {
+			return boolFalseResultContainer, err
+		}
+
+		return resultContainer{BoolValue: right.MatchString(left), IsBoolValue: true}, nil
+
+	},
+
+	// -----------------------------------------
+	//	Math evaluation funcs
+
+	operationKey{operator: tick.TokenPlus, leftType: TFloat64, rightType: TFloat64}: func(scope *tick.Scope, executionAux ExecutionAux, leftNode, rightNode NodeValueAccessor) (resultContainer, error) {
+		var left float64
+		var right float64
+		var err error
+
+		if left, err = leftNode.Float64Value(scope, executionAux); err != nil {
+			panic("harosa")
```

Comment:
=X my mistake, "panic(harosa)" is something I wrote to make sure I am getting to the code.

It will fix these and make sure there is test to cover this case.